### PR TITLE
Fixes #84: update to have one executor process a job

### LINKED ISSUE #84: Multiple migrators running the same set of jobs will duplicate work
This is because a migrator will greedy fetch all jobs and keep them in memory. So even after a job is updated by another migrator, other instance would never get that information as it has already essentially been cached.

### PR DESCRIPTION
**Pull request by jewzaam**
If multiple migrator instances try to process a job the execution that was put in the array first wins.  All other executions are marked as complete and end date is set.  The indication that nothing was done is in the document counts.  There is no special flag or anything to indicate it didn't do any work other than those counts.

### CODE REVIEW

File: consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java [hunk 1]
```diff
@@ -260,38 +268,80 @@ public void run() {
...
+                currentRun.setActualEndDate(new Date());
+                saveJobDetails(jobExecutionPsn);
+            } else {
+                // just mark complete and set actual end date
+                markExecutionComplete(jobExecutionPsn);
+            }
+
+        } catch (RuntimeException e) {
+            // would be nice to reference DataType.DATE_FORMAT_STR in core..
             SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMdd'T'HH:mm:ss.SSSZ");
```
Comment (derek63) [inline consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java]: You can use the one that is in lightblue client ClientConstants.getDateFormat()

File: consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java [hunk 2]
```diff
@@ -260,38 +268,80 @@ public void run() {
...
+            LightblueResponse response = saveJobDetails(jobExecutionPsn);
 
-            Map<String, JsonNode> destinationDocuments = getDestinationDocuments(sourceDocuments);
+            boolean processJob = true;
 
-            List<JsonNode> documentsToOverwrite = getDocumentsToOverwrite(sourceDocuments, destinationDocuments);
+            // verify this job is the first active thread processing the thread.
+            // first extract processed data as MigrationJob
+            JsonNode node = response.getJson().path("processed");
+            try {
```
Comment (derek63) [inline consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java]: Should we pull out the processJob (between here and line 311) logic into a separate method?

File: consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java [hunk 3]
```diff
@@ -260,38 +268,80 @@ public void run() {
...
+                }
 
-            if (!documentsToOverwrite.isEmpty()) {
-                hasInconsistentDocuments = true;
+                // first incomplete job execution must be for this execution else we don't process it
+                int i = 0;
+                for (MigrationJobExecution execution : jobs[0].getJobExecutions()) {
+                    // if we find an execution that is not our pid but is active
+                    // in the array before ours, we do not get to process
+                    if (!execution.isCompletedFlag() && !pid.equals(execution.getPid())) {
```
Comment (dcrissman) [inline consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java]: How are we handling the case where a job has been idle for some period of time? Presumably the process had an error and needs to be attempted again. Maybe the process on the server has been restarted so the pid is no longer the same.

File: consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java (hunk 2 above)
Comment (jewzaam) [inline consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java]: done

File: consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java (hunk 1 above)
Comment (jewzaam) [inline consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java]: done

File: consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java (hunk 3 above)
Comment (jewzaam) [inline consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java]: done, hard coded to time out job execution after 60 minutes

**Pull request merged**